src/main/kotlin/com/openlattice/transporter/TransporterQueries.kt

```diff
@@ -0,0 +1,298 @@
+package com.openlattice.transporter
+
+import com.openlattice.ApiUtil
+import com.openlattice.IdConstants
+import com.openlattice.postgres.PostgresArrays
+import com.openlattice.postgres.PostgresColumn.DST_ENTITY_KEY_ID
+import com.openlattice.postgres.PostgresColumn.DST_ENTITY_SET_ID
+import com.openlattice.postgres.PostgresColumn.EDGE_ENTITY_KEY_ID
+import com.openlattice.postgres.PostgresColumn.EDGE_ENTITY_SET_ID
+import com.openlattice.postgres.PostgresColumn.ENTITY_SET_ID
+import com.openlattice.postgres.PostgresColumn.ID_VALUE
+import com.openlattice.postgres.PostgresColumn.LAST_TRANSPORT
+import com.openlattice.postgres.PostgresColumn.LINKING_ID
+import com.openlattice.postgres.PostgresColumn.ORIGIN_ID
+import com.openlattice.postgres.PostgresColumn.PARTITION
+import com.openlattice.postgres.PostgresColumn.PROPERTY_TYPE_ID
+import com.openlattice.postgres.PostgresColumn.SRC_ENTITY_KEY_ID
+import com.openlattice.postgres.PostgresColumn.SRC_ENTITY_SET_ID
+import com.openlattice.postgres.PostgresColumn.VERSION
+import com.openlattice.postgres.PostgresColumnDefinition
+import com.openlattice.postgres.PostgresExpressionIndexDefinition
+import com.openlattice.postgres.PostgresTable
+import com.openlattice.postgres.PostgresTable.E
+import com.openlattice.postgres.PostgresTableDefinition
+import com.openlattice.transporter.types.TransporterColumn
+import com.zaxxer.hikari.HikariDataSource
+import org.slf4j.Logger
+import java.sql.Connection
+import java.util.*
+
+const val transporterNamespace = "transporter_data"
+
+private val transportTimestampColumn: PostgresColumnDefinition = LAST_TRANSPORT
+
+private const val BATCH_LIMIT = 10_000
+
+val MAT_EDGES_TABLE = edgesTableDefinition()
+
+fun unquotedTableName(entityTypeId: UUID): String {
+    return "et_$entityTypeId"
+}
+
+fun tableName(entityTypeId: UUID): String {
+    return ApiUtil.dbQuote(unquotedTableName(entityTypeId))
+}
+
+fun edgesTableDefinition(): PostgresTableDefinition {
+    val definition = PostgresTableDefinition("edge_assemblies")
+    definition.addColumns(
+            SRC_ENTITY_SET_ID,
+            SRC_ENTITY_KEY_ID,
+            DST_ENTITY_SET_ID,
+            DST_ENTITY_KEY_ID,
+            EDGE_ENTITY_SET_ID,
+            EDGE_ENTITY_KEY_ID
+    )
+    definition.primaryKey(
+            SRC_ENTITY_SET_ID,
+            SRC_ENTITY_KEY_ID,
+            DST_ENTITY_SET_ID,
+            DST_ENTITY_KEY_ID,
+            EDGE_ENTITY_SET_ID,
+            EDGE_ENTITY_KEY_ID
+    )
+    return definition
+}
+
+fun tableDefinition(entityTypeId: UUID, propertyColumns: Collection<PostgresColumnDefinition>): PostgresTableDefinition {
+    val definition = PostgresTableDefinition(tableName(entityTypeId))
+    val indexPrefix = unquotedTableName(entityTypeId) + "_"
+    definition.addColumns(
+            ENTITY_SET_ID,
+            ID_VALUE,
+            ORIGIN_ID
```

Comment:
is ORIGIN_ID actually LINKING_ID?

Reply:
/how is linking data handled here generally?

Reply:
I think this can probably be omitted right now actually. There's really nothing linky being done